### PR TITLE
FIX: topomap with grad ignored sphere and used layout (#7404)

### LINKED ISSUE #7404: BUG topomap grads
```
import numpy as np
import mne
from mne.datasets import sample
data_path = sample.data_path()
raw = mne.io.read_raw_fif(data_path + '/MEG/sample/sample_audvis_raw.fif')
info = raw.load_data().pick('grad').info
data = np.random.randn(204)
mne.viz.plot_topomap(data, info)
```

produces 
Figure_1

### PR DESCRIPTION
**Pull request by mmagnuski**
@agramfort - see if this works for you (it does for me).
fixes #7404 #7406

###  TODO:
- [x] add test
- [x] no need for whatsnew update, right?

### CODE REVIEW

**Comment by mmagnuski** [comment]
no need for a test?

**Comment by mmagnuski** [comment]
The one failing travis job seems unrelated - errors are due to some deprecation warnings in mayavi/3d plotting.

**Comment by larsoner** [comment]
Maybe the thing to do is use @agramfort's code snippet, do the plot, and check that the sensor locations in the plot (via `fig.axes[0].lines[0].get_xdata()` or whatever) are not too far from the xmin/xmax using `axes.get_xlim` (and same for Y). It should fail on `master` and pass here I think.

**Comment by agramfort** [comment]
whatever is doable with small pain. I can live without a test here. Yes I
know I am getting old :)

>

**Comment by larsoner** [comment]
I leave it up to you @mmagnuski let's see how strong the intrinsic motivation is for you on this one :)

**Comment by larsoner** [comment]
... and yes no need for `latest.inc` update because this is a within-cycle fix

**Comment by mmagnuski** [comment]
Haha, I have small but significant motivation to write a test, I'll add it today. :)

File: mne/viz/tests/test_topomap.py [hunk 1]
```diff
@@ -234,6 +234,18 @@ def test_plot_topomap_basic():
     with pytest.raises(ValueError, match=error_msg):
         plot_topomap(data, info, extrapolate='head', border='fancy')
 
+    # test channel placement when only 'grad' are picked:
+    # ---------------------------------------------------
+    info_grad = evoked.copy().pick('grad').info
+    n_grads = len(info_grad['ch_names'])
+    data = np.random.randn(n_grads)
+    img, cnt = plot_topomap(data, info_grad)
```
Comment (agramfort) [inline mne/viz/tests/test_topomap.py]:
```suggestion
    fig, _= plot_topomap(data, info_grad)
```

File: mne/viz/tests/test_topomap.py [hunk 2]
```diff
@@ -234,6 +234,18 @@ def test_plot_topomap_basic():
...
 
+    # test channel placement when only 'grad' are picked:
+    # ---------------------------------------------------
+    info_grad = evoked.copy().pick('grad').info
+    n_grads = len(info_grad['ch_names'])
+    data = np.random.randn(n_grads)
+    fig, _ = plot_topomap(data, info_grad)
+
+    # check that channels are scattered around x == 0
+    pos = img.axes.collections[-1].get_offsets()
```
Comment (larsoner) [inline mne/viz/tests/test_topomap.py]: This name needs to change, too, right?

File: mne/viz/tests/test_topomap.py [hunk 3]
```diff
@@ -242,7 +242,7 @@ def test_plot_topomap_basic():
     fig, _ = plot_topomap(data, info_grad)
 
     # check that channels are scattered around x == 0
-    pos = img.axes.collections[-1].get_offsets()
+    pos = fig.axes.collections[-1].get_offsets()
```
Comment (mmagnuski) [inline mne/viz/tests/test_topomap.py]: @larsoner Well, technically, this is actually not `fig`, but `img`, I read @agramfort suggestion too quickly. :)

**Comment by agramfort** [comment]
CIs error are unrelated. Merging. thx @mmagnuski

**Comment by mmagnuski** [comment]
Thanks @agramfort @larsoner ! :rocket:

**Pull request merged**